state-chain/pallets/cf-swapping/src/lib.rs

```diff
@@ -698,6 +675,94 @@ pub mod pallet {
 
 			Ok(())
 		}
+
+		/// Request a swap deposit address.
+		///
+		/// ## Events
+		///
+		/// - [SwapDepositAddressReady](Event::SwapDepositAddressReady)
+		#[pallet::call_index(10)]
+		#[pallet::weight(T::WeightInfo::request_swap_deposit_address())]
+		pub fn request_swap_deposit_address_v2(
+			origin: OriginFor<T>,
+			source_asset: Asset,
+			destination_asset: Asset,
+			destination_address: EncodedAddress,
+			broker_commission: BrokerFees<T::AccountId>,
+			channel_metadata: Option<CcmChannelMetadata>,
+			boost_fee: BasisPoints,
+		) -> DispatchResult {
+			ensure!(T::SafeMode::get().deposits_enabled, Error::<T>::DepositsDisabled);
+			let broker = T::AccountRoleRegistry::ensure_broker(origin)?;
+			let (beneficiaries, total_bps): (
+				BoundedVec<Beneficiary<T::AccountId>, ConstU32<MAX_BENEFICIARIES>>,
+				BasisPoints,
+			) = match broker_commission {
+				BrokerFees::Single(broker_bps) =>
+					if broker_bps > 0 {
+						(
+							vec![Beneficiary { account: broker.clone(), bps: broker_bps }]
+								.try_into()
+								.unwrap(),
+							broker_bps,
+						)
+					} else {
+						(vec![].try_into().unwrap(), broker_bps)
+					},
+				BrokerFees::Multiple(ref fees_bps) => {
+					for Beneficiary { account, bps: _ } in fees_bps.clone() {
+						ensure!(
+							T::AccountRoleRegistry::has_account_role(&account, AccountRole::Broker),
+							Error::<T>::BeneficiaryAccountIsNotABroker
+						);
+					}
+					(
+						(*fees_bps)
+							.clone()
+							.into_iter()
+							.filter(|Beneficiary { account: _, bps }| *bps > 0)
+							.collect::<Vec<_>>().try_into().expect("The provided beneficiaries vec cannot have more than MAX_BENEFICIARIES elements"),
+						fees_bps.iter().fold(0, |acc, Beneficiary { account: _, bps }| acc + bps),
```

Comment:
I am not sure if there is a better way to do this but we are pretty sure this conversion can never fail since the argument passed to the extrinsic is already a BoundedVec.